libs/community/langchain_community/document_loaders/upstage.py

```diff
@@ -0,0 +1,227 @@
+import os
+from enum import Enum
+from typing import List
+
+import requests
+from langchain_core.documents import Document
+
+from langchain_community.document_loaders.base import BaseLoader
+
+LAYOUT_ANALYZER_URL = "https://api.upstage.ai/v1/document-ai/layout-analyzer"
+
+
+class OutputType(Enum):
+    """
+    Represents the output type for a document loader.
+    """
+
+    TEXT = "text"
+    HTML = "html"
+
+
+class SplitType(Enum):
+    """
+    Enum class representing the type of split for a document.
+
+    Attributes:
+        NONE (str): Represents no split.
+        ELEMENT (str): Represents splitting by element.
+        PAGE (str): Represents splitting by page.
+    """
+
+    NONE = "none"
+    ELEMENT = "element"
+    PAGE = "page"
+
+
+def validate_api_key(api_key: str) -> None:
+    """
+    Validates the provided API key.
+
+    Args:
+        api_key (str): The API key to be validated.
+
+    Raises:
+        ValueError: If the API key is empty or None.
+
+    Returns:
+        None
+    """
+    if not api_key:
+        raise ValueError("API Key is required for Upstage Document Loader")
+
+
+def validate_file_path(file_path: str) -> None:
+    """
+    Validates if a file exists at the given file path.
+
+    Args:
+        file_path (str): The path to the file.
+
+    Raises:
+        FileNotFoundError: If the file does not exist at the given file path.
+    """
+    if not os.path.exists(file_path):
+        raise FileNotFoundError(f"File not found: {file_path}")
+
+
+def parse_output(data: dict, output_type: str) -> str:
+    """
+    Parse the output data based on the specified output type.
+
+    Args:
+        data (dict): The data to be parsed.
+        output_type (str): The type of output to parse.
+
+    Returns:
+        str: The parsed output.
+
+    Raises:
+        ValueError: If the output type is invalid.
+    """
+    if (output_type) == OutputType.TEXT.value:
+        return data["text"]
+    elif (output_type) == OutputType.HTML.value:
+        return data["html"]
+    else:
+        raise ValueError(f"Invalid output type: {output_type}")
+
+
+class UpstageDocumentLoader(BaseLoader):
+    """
+    A document loader for Upstage API.
+
+    Args:
```

Comment:
We commonly put the doc-string for the init arguments in the __init__ section.

Our API reference docs will concatenate both sections if I'm not mistaken

Reply:
I fix the code with your comment